plugins/discourse-adplugin/app/controllers/ad_plugin/ad_impressions_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+module AdPlugin
+  class AdImpressionsController < ::ApplicationController
+    requires_plugin PLUGIN_NAME
+
+    skip_before_action :check_xhr, :preload_json, :verify_authenticity_token, only: [:update]
```

Comment:
Instead of skipping the authenticity check here, can we send `X-CSRF-Token` in header? We implemented something similar in [this PR](https://github.com/discourse/discourse-page-visits/pull/28).

Here's the [link to specific code added](https://github.com/discourse/discourse-page-visits/pull/28/files#diff-e2cf06a3984803850a1a1effabc8d24020debfa745eed9389653213d85133fc0R145-R149).